Lib/zipfile/__init__.py

```diff
@@ -508,6 +518,24 @@ def _decodeExtra(self):
                 except struct.error:
                     raise BadZipFile(f"Corrupt zip64 extra field. "
                                      f"{field} not found.") from None
+            elif tp == 0x7075:
+                data = extra[4:ln+4]
+                # Unicode Path Extra Field
+                try:
+                    up_version, up_name_crc = unpack('<BL', data[:5])
+                    if up_version == 1 and up_name_crc == self.orig_filename_crc:
+                        up_unicode_name = data[5:].decode('utf-8')
```

Comment:
According the the spec, please check the TSize if possible.
````

As UnicodeName is defined to be UTF-8, no UTF-8
byte order mark (BOM) is used.  The length of this field is determined
by subtracting the size of the previous fields from TSize.
````

Reply:
The TSize is stored in the `ln` variable. I don't think we need to check it because we are storing data of the size defined by `ln`. If there is a problem, an exception would be thrown.